avaframe/ana3AIMEC/aimecTools.py

```diff
@@ -543,13 +502,20 @@ def getSArea(rasterTransfo):
 
     area = np.abs(x1*y2-y1*x2 + x2*y3-y2*x3 + x3*y4-y3*x4 + x4*y1-y4*x1)/2
 
-    # Method 2
-    # calculate area of each cell assuming they are parallelogramms
-    # (which is wrong)
-    # newAreaRaster = np.abs(dxl*dys - dxs*dyl)
-
     # save Area matrix
-    rasterTransfo['rasterArea'] = area
+    demCellSize = dem['header']['cellsize']
+    xllcenter = dem['header']['xllcenter']
+    yllcenter = dem['header']['yllcenter']
+    # area corection coef due to slope (1 if cell is horizontal, >1 if sloped)
+    demAreaCoef = dem['areaRaster']/(demCellSize*demCellSize)
+    # project on ney grid
+    areaCoef, _ = geoTrans.projectOnGrid(rasterTransfo['gridx']*demCellSize, rasterTransfo['gridy']*demCellSize,
+                                            demAreaCoef, csz=demCellSize, xllc=xllcenter, yllc=yllcenter)
+    newRasterRealArea = area * areaCoef
+    # projected area
+    # rasterTransfo['rasterArea'] = area
+    # real area
+    rasterTransfo['rasterArea'] = newRasterRealArea
```

Comment:
I would combine this line with line 514 and remove the variable newRasterRealArea...